### PR TITLE
node: signatures of module "querystring" have been changed

### PR DESCRIPTION
**Pull request by chrootsu**
https://nodejs.org/dist/v4.2.3/docs/api/querystring.html

### CODE REVIEW

File: node/node.d.ts [hunk 1]
```diff
@@ -405,8 +405,19 @@ declare module "buffer" {
...
+        encodeURIComponent?: Function;
+    }
+
+    export interface ParseOptions {
+        maxKeys?: number;
+        decodeURIComponent?: Function;
+    }
+
+    export function stringify(obj: any, sep?: string, eq?: string, options?: StringifyOptions): string;
+    export function stringify<T extends {}>(obj: T, sep?: string, eq?: string, options?: StringifyOptions): string;
```
Comment (vvakame) [inline node/node.d.ts]: Why ` extends {}` is needed?

**Comment by chrootsu** [comment]
@vvakame Oops, you absolutely right, it should be `any`. The function `stringify` returns an empty string if the argument `obj` is not Object... Thanks mate! Fixed. =)

File: node/node.d.ts [hunk 2]
```diff
@@ -405,8 +405,18 @@ declare module "buffer" {
...
+    }
+
+    export interface ParseOptions {
+        maxKeys?: number;
+        decodeURIComponent?: Function;
+    }
+
+    export function stringify<T>(obj: T, sep?: string, eq?: string, options?: StringifyOptions): string;
+    export function parse(str: string, sep?: string, eq?: string, options?: ParseOptions): any;
+    export function parse<T extends {}>(str: string, sep?: string, eq?: string, options?: ParseOptions): T;
```
Comment (vvakame) [inline node/node.d.ts]: same as above :wink:

Comment (chrootsu) [inline node/node.d.ts]: Hmm... I didn't find any case when the function `parse` returns any type except a type `Object`:
```
> querystring.parse('a=1')
{ a: '1' }
> querystring.parse('a=')
{ a: '' }
> querystring.parse('=1')
{ '': '1' }
> querystring.parse('')
{}
> querystring.parse(1)
{}
```
even if it used with a custom `decodeURIComponent`:
```
> querystring.parse('a=1', null, null, {decodeURIComponent: x => true})
{ true: true }
> querystring.parse('a=1', null, null, {decodeURIComponent: () => {}})
{ undefined: undefined }

```
Are you sure about it?

Comment (vvakame) [inline node/node.d.ts]: ah, okey.

**Comment by vvakame** [comment]
thanks mate!

**Pull request merged**